### PR TITLE
Resend pingback after successfull and wait for the most recent auth request to resolve analytics values

### LINKED ISSUE #2340: AMP Access extension - No pingback fired after login
Our testing shows that after a user logs in, the AMP Access extensions triggers an amp-access request.
The user is able to see the content, but there is no pingback request fired by the framework.

This will affect metered models were AMP devices will be an extra view.

### PR DESCRIPTION
**Pull request by dvoytenko**
Two important changes in spec:

1. Pingback will be resend after successful login to signify a new view with the new access profile.
2. The analytics values and pingback wait for the most recent authorization call to complete, but do tolerate some failures.

Closes #2340

/cc @ashwinlimaye @rudygalfi @sebastianbenz @andreban @adaybujeda @georgecrawford

### CODE REVIEW

File: extensions/amp-access/0.1/amp-access.js [hunk 1]
```diff
@@ -394,6 +401,10 @@ export class AccessService {
       this.toggleTopClass_('amp-access-loading', false);
       this.toggleTopClass_('amp-access-error', true);
     });
+    // At least one promise should succeed.
```
Comment (cramforce) [inline extensions/amp-access/0.1/amp-access.js]: The comment indicates you want `.race`?

File: extensions/amp-access/0.1/amp-access.js [hunk 2]
```diff
@@ -599,10 +619,16 @@ export class AccessService {
   /**
    * The promise will be resolved when a view of this document has occurred. It
    * will be rejected if the current impression should not be counted as a view.
+   * @param {time} timeToView
    * @return {!Promise}
    * @private
    */
-  whenViewed_() {
+  whenViewed_(timeToView) {
+    if (timeToView == 0) {
```
Comment (cramforce) [inline extensions/amp-access/0.1/amp-access.js]: Please document when one would use 0.

File: extensions/amp-access/0.1/amp-access.js (hunk 1 above)
Comment (dvoytenko) [inline extensions/amp-access/0.1/amp-access.js]: Updated comment. The right wording "first promise is always required to complete".

File: extensions/amp-access/0.1/amp-access.js (hunk 2 above)
Comment (dvoytenko) [inline extensions/amp-access/0.1/amp-access.js]: Documented. The reason for 0 is when this method is called as the result of the user action.

**Comment by dvoytenko** [comment]
@cramforce PTAL

**Comment by cramforce** [comment]
LGTM

**Pull request merged**